testsuite/classlibrary/TestLinkClock.sc

```diff
@@ -202,4 +202,140 @@ TestLinkClock : UnitTest {
 
 		semaphore.wait
 	}
+
+	test_LinkClock_meter_change_beats_keep_continuity {
+		var clock1, clock2,
+		tempo = 12, beatDurThresh = 0.4 / tempo,
+		secList1 = List.new, secList2 = List.new,
+		cond = Condition({ secList1.size == 20 and: { secList2.size == 20 } });
+
+		clock1 = LinkClock(tempo);
+		{
+			2.5.wait;  // offset clock starts
+			clock2 = LinkClock(tempo);
+			1.5.wait;
+			{
+				20.do {
+					secList2.add(clock2.seconds);
+					1.0.wait;
+				};
+				cond.signal;
+			}.fork(clock2);
+			20.do {
+				secList1.add(clock1.seconds);
+				clock1.setMeterAtBeat(rrand(2, 4), clock1.beats);
+				1.0.wait;
+			};
+			cond.signal;
+		}.fork(clock1);
+		cond.wait;
+		this.ifAsserts(
+			clock1.numPeers == 1,
+			"no extra clocks in session",
+			{
+				this.assert(
+					(secList1 absdif: secList2).every(_ <= beatDurThresh),
+					"LinkClock:setMeterAtBeat should not disrupt beat duration\nclock1: %\nclock2: %\n"
+					.format(secList1, secList2)
+				)
+			},
+			{
+				this.assert(false, "Make sure no other LinkClocks are running while testing");
```

Comment:
in general, we assume that users are not running other servers or sc processes while testing. guarding against every possible case of that would just be took much work. i'd prefer not to have these checks, but maybe i'm missing something. could this be affected by people running LinkClocks on the local network on another machine?

Reply:
> could this be affected by people running LinkClocks on the local network on another machine?

Yes, it will be. I do understand the general assumption but LinkClock strikes me as a special case, as it's designed to seek out peers actively on the LAN.

Will go over these comments in the next couple of days.